huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_resource.go

```diff
@@ -0,0 +1,285 @@
+		ReadContext:   resourceStudioResourceRead,
+		UpdateContext: resourceStudioResourceUpdate,
+		DeleteContext: resourceStudioResourceDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceStudioResourceImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"workspace_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"directory": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"location": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"depend_packages": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"location": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceStudioResourceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	//nolint:misspell
+	createResourceHttpUrl := "v1/{project_id}/resources"
+	createResourceProduct := "dataartsv1"
+
+	createResourceClient, err := cfg.NewServiceClient(createResourceProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+	createResourcePath := createResourceClient.Endpoint + createResourceHttpUrl
+	createResourcePath = strings.ReplaceAll(createResourcePath, "{project_id}", createResourceClient.ProjectID)
+
+	createResourceOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+	}
+	createResourceOpt.JSONBody = utils.RemoveNil(buildCreateOrUpdateResourceBodyParams(d))
+	createResourceResp, err := createResourceClient.Request("POST", createResourcePath, &createResourceOpt)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio resource: %s", err)
+	}
+
+	createResourceRespBody, err := utils.FlattenResponse(createResourceResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := utils.PathSearch("resourceId", createResourceRespBody, nil)
+	if id == nil {
+		return diag.Errorf("error creating DataArts Studio resource: %s is not found in API response", "id")
+	}
+	d.SetId(id.(string))
+	return resourceStudioResourceRead(ctx, d, meta)
+}
+
+func buildCreateOrUpdateResourceBodyParams(d *schema.ResourceData) map[string]interface{} {
+	var dependPackages []map[string]string
+
+	if rawParams, ok := d.Get("depend_packages").([]interface{}); ok {
```

Comment:
it's better to use d.GetOk

```suggestion
	if v, ok := d.GetOk("depend_packages"); ok {
		rawParams := v.([]interface{})
```